sherpa/ui/utils.py

```diff
@@ -10412,6 +10412,73 @@ def get_draws(self, id=None, otherids=(), niter=1000, covar_matrix=None):
             fit, covar_matrix, niter=niter)
         return (stats, accept, params)
 
+    def resample_data(self, id=None, niter=1000, seed=123):
```

Comment:
I think there's a possibility for user confusion here. `resample_data` is defined in `sherpa.ui` (and so will also be available to users of `sherpa.astro.ui`) but `load_ascii_with_errors` is only available to users of `sherpa.astro.ui` (because of it's use of `sherpa.astro.io`).

Can we make a version of `load_ascii_with_errors` that can be used from `sherpa.ui` - or are existing I/O functions for this module not sufficient? (There is also the fact that I think the way we extend functionality in `sherpa.astro.ui` by copying code is not ideal but that's a different PR...).